tests/webTables.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await expect(page.locator("h2")).toHaveText("Specialties");
+
+    const rowSurgery = page.locator('tr', { has: page.locator('[id="1"]') })
+    await rowSurgery.getByRole("button", { name: "Edit" }).click();
+    await expect(page.locator('h2')).toContainText('Edit Specialty')
+    const editSpecialtyInput = page.locator('#name')
+    await editSpecialtyInput.click()
+    await editSpecialtyInput.clear()
+    await editSpecialtyInput.fill('dermatology')
+    await page.getByRole('button', { name: "Update" }).click()
+    await expect(rowSurgery.locator('td input').first()).toHaveValue('dermatology')
+
+    await veterinariansMenu.click();
+    await veterinariansMenuAll.click();
+    expect(await rafaelSpecialty.textContent()).toContain('dermatology')
+
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await rowSurgery.getByRole("button", { name: "Edit" }).click();
+    await editSpecialtyInput.click()
+    await editSpecialtyInput.clear()
+    await editSpecialtyInput.fill('surgery')
+    await page.getByRole('button', { name: "Update" }).click()
+})
+
+test('Validate specialty lists', async ({ page }) => {
+    await page.goto("/");
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await page.getByRole('button', { name: "Add" }).click()
+    await page.locator('#name').fill('oncology')
+    await page.getByRole('button', { name: "Save" }).click()
+
+    const allSpecialtiesValues = await page.locator('td input')
+    let specialtiesList = []
+
+    for (const specialty of await allSpecialtiesValues.all()) {
+        specialtiesList.push(await specialty.inputValue())
+    }
+
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
+    const sharonJenkinsRow = page.locator('tr', { has: page.getByText(' Sharon Jenkins ') })
+    await sharonJenkinsRow.getByRole('button', { name: 'Edit Vet' }).click()
+
+    await page.locator('.dropdown-display').click()
+    let specialtiesListFromDropDOwn = []
+    const allValuesFromDropDownMenu = await page.locator(".dropdown-content label").allTextContents()
+    specialtiesListFromDropDOwn.push(allValuesFromDropDownMenu)
+    expect(specialtiesList).toBe(specialtiesList)
+
+    await page.getByLabel('oncology').click()
+    await page.locator('.selected-specialties').click()
+    await page.getByRole('button', { name: 'Save Vet' }).click()
+    expect(await sharonJenkinsRow.locator('td').nth(1).textContent()).toContain('oncology')
+
+    await page.getByRole('link', { name: 'Specialties' }).click()
+    await page.locator('tbody tr').last().getByRole('button', { name: "Delete" }).click()
+
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
+    expect(await sharonJenkinsRow.locator('td').nth(1).textContent()).toContain('')
```

Comment:
There is a better Locator Assertion available to validate that input field is empty